### PR TITLE
media-libs/x265: fix SRC_URI and HOMEPAGE

### PR DESCRIPTION
**Pull request by ngg**
x265's source code was migrated from Mercurial to Git, this change
reflects that. I've also synced the contents between the 9999
ebuild and the 3.4 versions and updated the 9999 ebuild's subslot based
on the current .so name of the master branch.

Closes: https://bugs.gentoo.org/739798
Signed-off-by: Gergely Nagy <ngg@ngg.hu>

### CODE REVIEW

**Comment by ConiKost** [comment]
Please rebase.

**Review by aballier — APPROVED** [review]

**Comment by ngg** [comment]
I've rebased to current master without any conflicts. Hopefully it re-triggers CI check.

**Comment by ngg** [comment]
Could you merge this PR?

**Comment by achea** [comment]
I have been unable to build this stable package because the downloads 404:

```
--2020-10-16 12:54:15--  https://bitbucket.org/multicoreware/x265/downloads/x265_3.4.tar.gz
Resolving bitbucket.org... 18.205.93.1, 18.205.93.2, 18.205.93.0, ...
Connecting to bitbucket.org|18.205.93.1|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2020-10-16 12:54:15 ERROR 404: Not Found.


!!! Couldn't download 'x265_3.4.tar.gz'. Aborting.
 * Fetch failed for 'media-libs/x265-3.4', Log file:
```

File: media-libs/x265/x265-3.4.ebuild [hunk 1]
```diff
@@ -6,15 +6,16 @@ EAPI=7
...
 
 if [[ ${PV} = 9999* ]]; then
-	inherit mercurial
-	EHG_REPO_URI="https://bitbucket.org/multicoreware/x265"
+	inherit git-r3
+	EGIT_REPO_URI="https://bitbucket.org/multicoreware/x265_git/"
+	S=${WORKDIR}/${P}/source
 else
-	SRC_URI="https://bitbucket.org/multicoreware/x265/downloads/${PN}_${PV}.tar.gz"
+	SRC_URI="https://bitbucket.org/multicoreware/x265_git/get/${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (winterheart) [inline media-libs/x265/x265-3.4.ebuild:13]: Official repo of x265 remains be as a mercurual repository located here: http://hg.videolan.org/x265

File: media-libs/x265/x265-3.4.ebuild [hunk 2]
```diff
@@ -6,15 +6,16 @@ EAPI=7
...
 	KEYWORDS="amd64 arm arm64 hppa ~ia64 ppc ppc64 x86"
 fi
 
 DESCRIPTION="Library for encoding video streams into the H.265/HEVC format"
-HOMEPAGE="http://x265.org/ https://bitbucket.org/multicoreware/x265/wiki/Home"
+HOMEPAGE="http://x265.org/ https://bitbucket.org/multicoreware/x265_git/wiki/Home"
```
Comment (winterheart) [inline media-libs/x265/x265-3.4.ebuild:18]: HOMEPAGE should be http://x265.org/ and https://www.videolan.org/developers/x265.html

**Review by winterheart — CHANGES_REQUESTED** [review]

File: media-libs/x265/x265-3.4.ebuild (hunk 2 above)
Comment (jospezial) [reply media-libs/x265/x265-3.4.ebuild:18]: http://x265.org/ and https://www.videolan.org/developers/x265.html and http://hg.videolan.org/x265 are not uptodate.
https://x265.org expired certificate
https://bitbucket.org/multicoreware/x265_git/ is the only way to go at the moment.
Please stabilize media-libs/x265-3.5 and remove 3.4!

**Comment by ngg** [comment]
Fixed by https://github.com/gentoo/gentoo/commit/7788f2443c1b21a25dec1b996ac7af941926cb17

**Pull request closed without merging**